### PR TITLE
[RFC] doc: Make it easier to report issues in the documentation

### PR DESCRIPTION
**Pull request by kartben**
Live docs: https://builds.zephyrproject.io/zephyr/pr/61605/docs/

This is an attempt at fixing #60622 - please check there for context. In a nutshell: it adds to the bottom of each documentation page a note about when the most recent change to the page (i.e git commit) was done.

This is meant to give a high-level, probably not perfect but yet useful, indicator of the freshness of a given page and likeliness it might contain outdated content or mistakes. It's not perfect since it will sound very "pessimistic" for pages not meant to be updated often and that will show up with a very old date, or "optimistic" when it will indicate a very recent change to a page otherwise very old.

A note at the bottom of the page invites people to report inaccuracies/mistakes by opening a pre-filled GH issue.

image

The breadcrumb has also been reworked to include a direct link to open an issue.

image

I was initially look at adding an "Edit on Github" link, but it's probably not a good idea to have it as newcomers would still likely miss compliance, and people with write access to the repo would end up creating branches in the main repo as opposed to their own fork, as that's the default behavior of the Github web UI.

# TODO

- [x] Clean up the code :)
- [x] The intent would be to have the GitHub link create a pre-filled issue with as much information as possible, e.x SHA-1 of the file, path of course, etc.
- [ ] If deemed useful, I think the opened issue could be configured to set the area labels that match the doc page's path. Right now the only label being set is `area: Documentation`. This is not necessarily hard to implement, we just need to pull in a dependency to get_maintainer.py.
  
For the record this probably slows down the build of the documentation but doesn't seem too noticeable AFAICT.

### CODE REVIEW

**Comment by kartben** [comment]
Updated the code to actually add the right info when creating the issue ("area: Documentation" label, page name, sha-1 of the latest commit to the page).
Also updated the breadcrumb to add direct links to Open / Edit / Report Issue.

**Comment by carlescufi** [comment]
I think this is a very valuable change for our users wanting to contribute to documentation, please do continue the work so we can get it in.

**Comment by kartben** [comment]
@carlescufi @gmarull  -- I believe this is ready for review. 
FWIW I have no idea what the compliance check is complaining about, and I can't reproduce it locally...

File: doc/_extensions/zephyr/vcs_link.py [hunk 1]
```diff
@@ -60,22 +68,70 @@ def vcs_link_get_url(app: Sphinx, pagename: str) -> Optional[str]:
...
+    **Environment**
+
+    * Page: `{pagename}`
+    * Version: {app.config.vcs_link_version}
+    * SHA-1: {sha1}
+    """
+        )
+    )
+
+    return f"{app.config.vcs_link_base_url}/issues/new?title={title}&labels={labels}&body={body}"
```
Comment (gmarull) [inline doc/_extensions/zephyr/vcs_link.py:113]: nit: this extension has now become GH specific, so the "vcs" generic naming should be abandoned

File: doc/_extensions/zephyr/git_info.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    for pattern, prefix in app.config.vcs_link_prefixes.items():
+        if re.match(pattern, pagename):
+            found_prefix = prefix
+            break
+
+    orig_path = os.path.join(
+        app.config.ZEPHYR_BASE,
+        found_prefix,
+        app.env.project.doc2path(pagename, basedir=False),
+    )
```
Comment (kartben) [inline doc/_extensions/zephyr/git_info.py:59]: Mmmm I think I found why pylint is complaining 🙃
I just remembered that I copy-pasted this from vcs_link extension... Will update to make sure there's  no such duplication...

File: doc/_extensions/zephyr/git_info.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+    for exclude in app.config.vcs_link_exclude:
+        if re.match(exclude, pagename):
+            return None
+
+    found_prefix = ""
+    for pattern, prefix in app.config.vcs_link_prefixes.items():
+        if re.match(pattern, pagename):
+            found_prefix = prefix
+            break
```
Comment (kartben) [inline doc/_extensions/zephyr/git_info.py:53]: Mmmm I think I found why pylint is complaining 🙃
I just remembered that I copy-pasted this from vcs_link extension... Will update to make sure there's  no such duplication...

Comment (kartben) [reply doc/_extensions/zephyr/git_info.py:53]: @gmarull I am actually not sure where would be the best place to put this -- almost feels like all git/github extensions could maybe live in the same extension?

Comment (gmarull) [reply doc/_extensions/zephyr/git_info.py:53]: maybe create a single "gh_utils" extension

Comment (kartben) [reply doc/_extensions/zephyr/git_info.py:53]: Deal!

File: doc/_extensions/zephyr/vcs_link.py (hunk 1 above)
Comment (kartben) [reply doc/_extensions/zephyr/vcs_link.py]: refactored both "vcs" and "git" extension in the latest update, please have another look, thx!

**Review by carlescufi — APPROVED** [review]
Looks great, thanks!

**Review by gmarull — APPROVED** [review]

**Pull request merged**